components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/import-file-dialog.js

```diff
@@ -16,266 +16,193 @@
+                        var files = event.target.files;
+                        var file = files[0];
+                        var reader = new FileReader();
 
-                var treeContainer  = fileImport.find("div").filter("#fileTree")
-                fileBrowser = new FileBrowser({container: treeContainer, application:app, fetchFiles:true,
-                    showWorkspace:false});
+                        fileName = file.name;
 
-                fileBrowser.render();
-                this._fileBrowser = fileBrowser;
+                        var existsResponse = existFileInPath({
+                            configName: fileName
+                        });
 
-                //Gets the selected location from tree and sets the value as location
-                this.listenTo(fileBrowser, 'selected', function (selectedLocation) {
-                    selectedFilePath = selectedLocation;
-                    var pathAttributes = selectedLocation.split(self.pathSeparator);
-                    var fileName = _.last(pathAttributes);
+                        if (existsResponse.error === undefined) {
+                            if (existsResponse.exists) {
+                                alertError("A file already exist in workspace with selected name.");
+                                return;
+                            } else {
+                                reader.readAsText(file);
+                            }
+                        } else {
+                            alertError("Error in reading the file.");
+                        }
 
-                    if(selectedLocation && fileName.lastIndexOf(".siddhi") != -1){
-                        location.val(fileName);
+                        reader.onload = (function (reader) {
+                            return function () {
+                                var fileContent = reader.result;
+                                importConfiguration(fileContent);
+                            }
+                        })(reader);
                     }
-                });
 
-
-                fileImport.find("button").filter("#importButton").click(function () {
-
-                    var _location = selectedFilePath;
-                    if (_.isEmpty(fileImport.find("input").filter("#location"))) {
-                        openFileWizardError.text("Invalid Value for Location.");
-                        openFileWizardError.show();
-                        return;
+                    function existFileInPath(options) {
+                        var client = self.app.workspaceManager.getServiceClient();
+                        var data = {};
+                        var workspaceServiceURL = app.config.services.workspace.endpoint;
+                        var saveServiceURL = workspaceServiceURL + "/exists/workspace";
+                        var payload = "configName=" + btoa("workspace" + app
+                            .getPathSeperator() + options.configName);
+
```

Comment:
@erangatl  shall we remove the unnecessary spaces?

Reply:
Fixed with #1690